### PR TITLE
UCM/SHMAT: fixed resource leak

### PR DESCRIPTION
**Pull request by hoopoepg**
- fixed resource leak in shmat implementation
- minor optimization

### CODE REVIEW

File: src/ucm/event/event.c [hunk 1]
```diff
@@ -286,6 +286,12 @@ void *ucm_shmat(int shmid, const void *shmaddr, int shmflg)
             attach_addr -= attach_addr % SHMLBA;
         }
         ucm_dispatch_vm_munmap((void*)attach_addr, size);
+        ucs_recursive_spin_lock(&ucm_kh_lock);
+        iter = kh_get(ucm_ptr_size, &ucm_shmat_ptrs, (void*)attach_addr);
+        if (iter != kh_end(&ucm_shmat_ptrs)) {
+            kh_del(ucm_ptr_size, &ucm_shmat_ptrs, iter);
+        }
+        ucs_recursive_spin_unlock(&ucm_kh_lock);
```
Comment (yosefe) [inline src/ucm/event/event.c:294]: can we unite common code with ucm_shmdt()?

Comment (hoopoepg) [reply src/ucm/event/event.c]: united

File: src/ucm/event/event.c [hunk 2]
```diff
@@ -287,10 +303,7 @@ void *ucm_shmat(int shmid, const void *shmaddr, int shmflg)
         }
         ucm_dispatch_vm_munmap((void*)attach_addr, size);
         ucs_recursive_spin_lock(&ucm_kh_lock);
-        iter = kh_get(ucm_ptr_size, &ucm_shmat_ptrs, (void*)attach_addr);
-        if (iter != kh_end(&ucm_shmat_ptrs)) {
-            kh_del(ucm_ptr_size, &ucm_shmat_ptrs, iter);
-        }
+        ucm_shm_del_entry_from_khash((void*)attach_addr, NULL);
```
Comment (yosefe) [inline src/ucm/event/event.c:306]: can we move spin_lock inside ucm_shm_del_entry_from_khash as well?

Comment (hoopoepg) [reply src/ucm/event/event.c:308]: I think yes...
there is a call ucm_get_shm_seg_size which was inside of lock section, but it has own lock for multithread - should work fine

**Comment by hoopoepg** [comment]
@yosefe ok to merge?

**Pull request merged**